### ISSUE TITLE
`solve()` seems to return wrong result for a linear system 

### ISSUE THREAD

**Issue by johannesflake** [sympy/sympy#22133]
`solve()` seems to return a wrong solution `s` for the linear system of equations `eq`:
```python
import sympy
a,b,c,d,e,f,g,h = sympy.symbols("a,b,c,d,e,f,g,h")
eq = [-6*a-12*b-12*c-2,
d+e+2*f/3+g-1,
-2*d-2*e-4*f/3-2*h-2,
d+2*e+2*f+g-1,
-2*d-4*e-4*f-2*h-3]
s = sympy.solve(eq)
#  {a: -2*b - 2*c - 1/3, d: 2*f/3 - 2*g + h + 7/2, e: -4*f/3 + g - h - 5/2}
[e.subs(s) for e in eq]
# [0, 0, 2*g - 2*h - 4, g - h - 5/2, 0]
```

**Comment by oscarbenjamin** [comment]
Current master gives no solutions for this:
```python
In [1]: import sympy
   ...: a,b,c,d,e,f,g,h = sympy.symbols("a,b,c,d,e,f,g,h")
   ...: eq = [-6*a-12*b-12*c-2,
   ...: d+e+2*f/3+g-1,
   ...: -2*d-2*e-4*f/3-2*h-2,
   ...: d+2*e+2*f+g-1,
   ...: -2*d-4*e-4*f-2*h-3]
   ...: s = sympy.solve(eq)

In [2]: s
Out[2]: []
```
I think that was changed by #21883

No solutions looks correct to me:
```python
In [12]: linsolve(eq, [a, b, c, d, e, f, g, h])
Out[12]: ∅

In [13]: M, rhs = linear_eq_to_matrix(eq, [a,b,c,d,e,f,g,h])

In [14]: Maug = Matrix.hstack(M, rhs)

In [15]: Maug
Out[15]: 
⎡-6  -12  -12  0   0    0    0  0   2⎤
⎢                                    ⎥
⎢0    0    0   1   1   2/3   1  0   1⎥
⎢                                    ⎥
⎢0    0    0   -2  -2  -4/3  0  -2  2⎥
⎢                                    ⎥
⎢0    0    0   1   2    2    1  0   1⎥
⎢                                    ⎥
⎣0    0    0   -2  -4   -4   0  -2  3⎦

In [16]: Maug.rref()
Out[16]: 
⎛⎡1  2  2  0  0   0    0  0   0⎤                 ⎞
⎜⎢                             ⎥                 ⎟
⎜⎢0  0  0  1  0  -2/3  0  1   0⎥                 ⎟
⎜⎢                             ⎥                 ⎟
⎜⎢0  0  0  0  1  4/3   0  0   0⎥, (0, 3, 4, 6, 8)⎟
⎜⎢                             ⎥                 ⎟
⎜⎢0  0  0  0  0   0    1  -1  0⎥                 ⎟
⎜⎢                             ⎥                 ⎟
⎝⎣0  0  0  0  0   0    0  0   1⎦                 ⎠
```
(final row implies no solutions)

**Comment by oscarbenjamin** [comment]
Fix bisected to b13f8775d53f7a345457eb8ab8732c008d22aee3 from #21883

**Comment by smichr** [comment]
The implied value for `g - h` also indicates that there is no solution: if the 4th equation is 0 then the 3rd equation would be 1 when `g-h=5/2`.
